libsolidity/codegen/CompilerUtils.cpp

```diff
@@ -988,10 +989,9 @@ void CompilerUtils::convertType(
 			{
 			case DataLocation::Storage:
 			{
-				auto conversionImpl = [
-					typeOnStack = dynamic_pointer_cast<StructType const>(_typeOnStack.shared_from_this()),
-					targetType = dynamic_pointer_cast<StructType const>(targetType.shared_from_this())
-				](CompilerContext& _context) {
+				auto conversionImpl =
+					[typeOnStack = &typeOnStack, targetType = &targetType](CompilerContext& _context)
```

Comment:
The previous code used `_typeOnStack`

Reply:
@chriseth: Actually, this code is right.

* the outer-scope `typeOnStack` is a reference type (which cannot be captured in lambdas) whereas the inner `typeOnStack` is a pointer type.
* It's not using the `_typeOnStack` as the original code did before along with a dynamic_cast to StructType, as the outer scope's `typeOnStack` already has been cast to that type.

Reply:
It is confusing anyway...